### PR TITLE
Deprecate the 'local' parameter of `/_cat/indices`

### PR DESCRIPTION
**Pull request by boicehuang**
The cat indices APIs perform a ClusterStateAction then an IndicesStatsAction. They accept the ?local parameter and pass this to the ClusterStateAction but this parameter has no effect on the IndicesStatsAction. 

This commit deprecates the ?local parameter on this API so that it can be removed in 8.0.

Related #60718

### CODE REVIEW

**Comment by danhermann** [comment]
@elasticmachine ok to test

**Comment by danhermann** [comment]
@elasticmachine update branch

File: docs/reference/cat/indices.asciidoc
Comment (danhermann) [inline docs/reference/cat/indices.asciidoc:85]: This creates two entries for the `local` parameter, one pulled from the `common-parms` file and this one. I'll solicit some input from our documentation writers on how best to handle this.

File: server/src/main/java/org/elasticsearch/rest/action/cat/RestIndicesAction.java
Comment (danhermann) [inline server/src/main/java/org/elasticsearch/rest/action/cat/RestIndicesAction.java:72]:
```suggestion
    static final String LOCAL_DEPRECATED_MESSAGE = "The parameter [local] is deprecated and will be removed in a future release.";
```

**Review by danhermann — CHANGES_REQUESTED** [review]
@boicehuang, thanks for addressing this issue. I think we can get this merged with a couple minor changes as noted below.

File: docs/reference/cat/indices.asciidoc
Comment (danhermann) [reply docs/reference/cat/indices.asciidoc:85]: @jrodewig, do you have any suggestions on how to handle this documentation case where a parameter pulled in a from a common file needs to be marked as deprecated on a specific API?

File: rest-api-spec/src/main/resources/rest-api-spec/api/cat.indices.json
Comment (jrodewig) [inline rest-api-spec/src/main/resources/rest-api-spec/api/cat.indices.json:56]: Shouldn't this be the release in which we deprecate the parameter? That should be before 8.0.0.

File: docs/reference/cat/indices.asciidoc
Comment (jrodewig) [inline docs/reference/cat/indices.asciidoc:85]: @boicehuang @danhermann You can simply remove the `include` statement. Since this parameter is currently no-op, we can just expand the deprecation message. Note you'll need to update the deprecation release based on where we backport this.
```suggestion
`local`::
(Optional, boolean)
+
deprecated::[7.10.0,"This parameter does not affect the request. It will be removed in version 8.0.0."]
```

Comment (jrodewig) [reply docs/reference/cat/indices.asciidoc:85]: I left a suggestion here: https://github.com/elastic/elasticsearch/pull/62198#discussion_r492857985

File: rest-api-spec/src/main/resources/rest-api-spec/api/cat.indices.json
Comment (danhermann) [reply rest-api-spec/src/main/resources/rest-api-spec/api/cat.indices.json:56]: We typically update the version when backporting to a release branch and then come back and update the master branch.

File: docs/reference/cat/indices.asciidoc
Comment (danhermann) [reply docs/reference/cat/indices.asciidoc:85]: Ah, thank you, that makes sense. I see now that each include specifies a single tag from the file.

**Comment by danhermann** [comment]
@elasticmachine update branch

**Comment by jrodewig** [comment]
@elasticmachine update branch

**Comment by jrodewig** [comment]
@boicehuang Will you merge the latest changes from `elastic/master` into this branch? There are several CI tests which have failed. We'll need to get those tests to pass before merging this PR. Thanks!

**Comment by danhermann** [comment]
@boicehuang, could you enable the "Allow edits and access to secrets by maintainers" option on this PR? That will enable us to use some of our automated tooling to do things like merge in the master branch, etc. I'll re-review it shortly. From a quick glance, it looks like we'll be able to merge it.

**Comment by boicehuang** [comment]
@danhermann  ok

**Comment by danhermann** [comment]
@elasticmachine update branch

**Comment by boicehuang** [comment]
@danhermann sorry, try again

**Comment by jrodewig** [comment]
@elasticmachine test this please

**Comment by jrodewig** [comment]
@elasticmachine update branch

**Comment by danhermann** [comment]
@boicehuang, this is ready to merge with the one exception of the documentation line here:

https://github.com/elastic/elasticsearch/pull/62198/files#diff-3c6feaa818c49894e13b5d59e34607fa68f4f0d4002012cb2400e7ddf79dcc58R80

which should be:

`deprecated::[8.0.0,"This parameter does not affect the request. It will be removed in a future release."]`

**Comment by danhermann** [comment]
@boicehuang, note that it is the line the `indices.asciidoc` file that needs to be updated. Also, please merge master into this PR so that the tests will pass. Normally, one of us would do that for you, but we cannot on this PR since edits aren't allowed.

**Comment by danhermann** [comment]
Thanks, @boicehuang. I will get this merged.

**Pull request merged**
